File: historyscan-back/src/main/java/io/sch/historyscan/domain/contexts/codebase/find/FindCodeBase.java
package io.sch.historyscan.domain.contexts.codebase.find;

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureAPI;

import java.util.Optional;

@HexagonalArchitectureAPI
@FunctionalInterface
public interface FindCodeBase {
    Optional<CurrentCodeBase> from(String name);
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/contexts/codebase/find/FindCodeBase.kt
package io.sch.historyscan.domain.contexts.codebase.find

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureAPI
import java.util.*

@HexagonalArchitectureAPI
fun interface FindCodeBase {
    fun from(name: String): CurrentCodeBase?
}
